contracts/cw3-fixed-multisig/src/contract.rs

```diff
@@ -352,28 +353,36 @@ fn list_votes(
     limit: Option<u32>,
 ) -> StdResult<VoteListResponse> {
     let limit = limit.unwrap_or(DEFAULT_LIMIT).min(MAX_LIMIT) as usize;
-    let start = start_after.map(|s| Bound::ExclusiveRaw(s.into()));
-
-    let votes = BALLOTS
-        .prefix(proposal_id)
-        .range(deps.storage, start, None, Order::Ascending)
-        .take(limit)
-        .map(|item| {
-            item.map(|(addr, ballot)| VoteInfo {
-                proposal_id,
-                voter: addr.into(),
-                vote: ballot.vote,
-                weight: ballot.weight,
-            })
-        })
-        .collect::<StdResult<_>>()?;
+    let start_address = start_after.map(|addr| Addr::unchecked(addr));
+
+    let suffix = proposal_id.to_ne_bytes();
+    let ballots = BALLOTS.add_suffix(&suffix);
+    let addresses = match start_address {
+        Some(addr) => VOTER_ADDRESSES.iter_from(deps.storage, &addr)?.skip(1),
+        None => VOTER_ADDRESSES.iter(deps.storage).skip(0),
+    }
+    .take(limit);
```

Comment:
I think this causes a bug where not all votes are necessarily counted. If you naively take `limit` voters like this and iterate over them, mapping them to ballots, then for every miss you risk not counting one ballot. We should be taking `limit` ballots, not voters.